scripts/parse_live.py

```diff
@@ -125,6 +151,60 @@ def wrapper(*args, **kwds):
                 raise
     return wrapper
 
+def insert(cursor, r):
+    cursor.execute('''
+        INSERT INTO articles VALUES(%s, %s, %s)''', r['article'])
+    cursor.executemany('''
+        INSERT IGNORE INTO snippets VALUES(%s, %s, %s, %s)''',
+        r['snippets'])
+
+    # We can't allow data to be truncated for HTML snippets, as that can
+    # completely break the UI, so we detect truncation warnings and get rid
+    # of the corresponding data.
+    warnings = cursor.execute('SHOW WARNINGS')
+    truncated_snippets = []
+    for _, _, message in cursor.fetchall():
+        m = DATA_TRUNCATED_WARNING_RE.match(message)
+        if m is None:
+            # Not a truncation, ignore (it's already logged)
+            continue
+        # MySQL warnings index rows starting at 1
+        idx = int(m.groups()[0]) - 1
+        truncated_snippets.append((r['snippets'][idx][0],))
+    if len(truncated_snippets) < len(r['snippets']):
+        cursor.executemany('''
+            DELETE FROM snippets WHERE id = %s''', truncated_snippets)
+    else:
+        # Every single snippet was truncated, remove the article itself
+        cursor.execute('''DELETE FROM articles WHERE page_id = %s''',
+            (r['article'][0],))
+
+@with_max_exceptions
+def work_sentences(sentences):
```

Comment:
Also here, there's quite a bit of duplication between this and `work`. I suspect the main difference between the two is just that `work_sentences` needs to call `extract_from_sentences`, so this can be deduplicated.

Reply:
That's true. I’ll also incorporate it into `work` instead. :grinning: